be/etp-backend/src/main/java/com/psa/TradingEngine.java

```diff
@@ -13,145 +13,199 @@
-    if (side.equals("B")) {
-      sql.append("DESC");
-    } else {
-      sql.append("ASC");
+
+    private void updateOrder(OrderEntity order, int filled) throws SQLException {
+        String sql = "UPDATE public.order SET quantity_filled = quantity_filled + ?, status = ? WHERE id = ?";
+        try (PreparedStatement pstmt = db.prepareStatement(sql)) {
+            pstmt.setInt(1, filled);
+            pstmt.setString(2, (
+                    order.quantity_filled() +
+                            order.quantity_cancelled() +
+                            filled
+            ) >= order.quantity() ? "COMPLETE" : order.status());
+            pstmt.setString(3, order.id());
+            pstmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
     }
-    sql.append(", timestamp ASC");
-    try (Statement stmt = db.createStatement()) {
-      ResultSet rs = stmt.executeQuery(sql.toString());
-      List<OrderEntity> orders = new ArrayList<>();
-      while (rs.next()) {
-        orders.add(new OrderEntity(
-              rs.getString("id"),
-              rs.getString("status"),
-              rs.getString("broker"),
-              rs.getString("symbol"),
-              rs.getString("side"),
-              rs.getString("type"),
-              rs.getDouble("price"),
-              rs.getInt("quantity"),
-              rs.getInt("quantity_filled"),
-              rs.getInt("quantity_cancelled"),
-              rs.getTimestamp("timestamp").toInstant().atZone(ZoneId.systemDefault()).toEpochSecond() * 1000
-              ));
-      }
-      return orders;
-    } catch (SQLException e) {
-      throw new RuntimeException(e);
+
+    private void updateLastPrice(String symbol, double price) throws SQLException {
+        // Update the price if the symbol exists, if not - insert a new record
+        String sql = "INSERT INTO public.last_price (symbol, price) " +
+                "VALUES (?, ?) " +
+                "ON CONFLICT (symbol) DO UPDATE " +
+                "SET price = EXCLUDED.price";
+
+        try (PreparedStatement pstmt = db.prepareStatement(sql)) {
+            pstmt.setString(1, symbol);
+            pstmt.setDouble(2, price);
+
+            pstmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
     }
-  }
-
-  private void updateOrder(OrderEntity order, int filled) throws SQLException {
-    String sql = "UPDATE public.order SET quantity_filled = quantity_filled + ?, status = ? WHERE id = ?";
-    try (PreparedStatement pstmt = db.prepareStatement(sql)) {
-      pstmt.setInt(1, filled);
-      pstmt.setString(2, (
-            order.quantity_filled() +
-            order.quantity_cancelled() +
-            filled
-            ) >= order.quantity() ? "COMPLETE" : order.status());
-      pstmt.setString(3, order.id());
-      pstmt.executeUpdate();
-    } catch (SQLException e) {
-      throw new RuntimeException(e);
+
```

Comment:
Let's default the STATUS to OPEN, sorry I forgot to mentioned that. There will be a separate mechanism to update the Symbol Status.